src/rgw/rgw_rados.cc

```diff
@@ -6571,8 +6569,20 @@ int RGWRados::bucket_index_read_olh_log(const RGWBucketInfo& bucket_info, RGWObj
 	                auto& ref = bs->bucket_obj.get_ref();
 			ObjectReadOperation op;
 			cls_rgw_guard_bucket_resharding(op, -ERR_BUSY_RESHARDING);
-			return cls_rgw_get_olh_log(ref.pool.ioctx(), ref.obj.oid, op,
-						   key, ver_marker, olh_tag, log, is_truncated);
+
+                        rgw_cls_read_olh_log_ret log_ret;
+                        int op_ret = 0;
+			cls_rgw_get_olh_log(op, key, ver_marker, olh_tag, log_ret, op_ret); 
+                        bufferlist outbl;
+                        int r =  rgw_rados_operate(ref.pool.ioctx(), ref.obj.oid, &op, &outbl, null_yield);
+                        if (r < 0) {
+                          return r;
+                        }
+                        if (op_ret < 0) {
+                          return op_ret;
+                        }
+
+                        return r;
```

Comment:
needs to assign the `log` and `is_truncated` output parameters in RGWRados::bucket_index_read_olh_log(), ie:
```c++
  *log = std::move(log_ret.log);
  *is_truncated = log_ret.is_truncated;
  return r;
```